tests/contrib/scrapy/test_pipeline_cerberus.py

```diff
@@ -0,0 +1,147 @@
+                    TestItem: "tests.fixtures.validators.cerberus_test_schema"
+                }
+            },
+            [assert_in_cerberus_stats(TestItem), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: [cerberus_test_schema]}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, validators are in a list repr".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            ["spidermon/validation/fields/errors/missing_required_field"],
+            id="Missing required fields",
+        ),
+    ],
+)
+def test_stats_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x in pipe.stats.stats.get_stats()
+        assert casechecker(case)
+
+
+@pytest.mark.parametrize(
+    "item,settings,cases",
+    [
+        pytest.param(
+            TestItem({"url": "example.com"}),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing simple items without errors",
+        ),
+        pytest.param(
+            TreeItem(
+                {
+                    "quotes": {"author": "Vipul Gupta", "quote": "Life is vanilla"},
+                    "child": "https://example.com",
+                }
+            ),
+            {SETTING_CERBERUS: [cerberus_tree_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing nested items without errors",
+        ),
+    ],
+)
+def test_stats_not_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x not in pipe.stats.stats.get_stats()
+        assert casechecker(case)
+
+
+def test_stats_amounts_in_pipeline():
+    item = TestItem({"title": "ScrapingHub"})
+    settings = {SETTING_CERBERUS: [cerberus_test_schema]}
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    assert pipe.stats.stats.get_stats()["spidermon/validation/validators"] is 1
+    assert (
+        pipe.stats.stats.get_stats()[
+            "spidermon/validation/fields/errors/missing_required_field"
+        ]
+        is 1
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 4), reason="mock requires python3.4 or higher"
+)
+def test_validation_from_url(mocker):
+    mocked_get_contents = mocker.patch(
```

Comment:
It looks like the mocking did not work

Reply:
It didn't. But, will give it a shot again